pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/AuthorizationService.java

```diff
@@ -218,7 +218,7 @@ public CompletableFuture<Boolean> canProduceAsync(TopicName topicName, String ro
         if (!this.conf.isAuthorizationEnabled()) {
             return CompletableFuture.completedFuture(true);
         }
-        return provider.isSuperUser(role, authenticationData, conf).thenComposeAsync(isSuperUser -> {
+        return provider.isSuperUser(role, authenticationData, conf).thenCompose(isSuperUser -> {
```

Comment:
This is a breaking change to the interface. That's why we should change from `thenComposeAsync` to `thenComple`. Switching to `thenComposeAsync` with an executor is a possible acceptable solution so that we don't break the interface for custom implementations.